net/nat/ipv4_nat.c

```diff
@@ -88,6 +84,44 @@ ipv4_nat_outbound_internal(FAR struct net_driver_s *dev,
                            FAR struct ipv4_hdr_s *ipv4,
                            enum nat_manip_type_e manip_type);
 
+/****************************************************************************
+ * Inline Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: ipv4_nat_l4_hdrlen
+ *
+ * Description:
+ *   Get L4 header length
+ *
+ ****************************************************************************/
+
+static inline uint16_t ipv4_nat_l4_hdrlen(uint8_t proto)
+{
+#ifdef CONFIG_NET_TCP
```

Comment:
Maybe better to reorganise in switch/case and debug assert in default?

Reply:
Done!